### PR TITLE
Host and Participant's Look same as evalAI look

### PR DESCRIPTION
**Pull request by Kajol-Kumari**
Features: 
Look of Participant's Page and Host Page of EvalAI-ngx made similar to EvalAI

GIF of Features added:
ezgif com-video-to-gif (17)

GIF for collapsible fix:
ezgif com-video-to-gif (1)

### CODE REVIEW

**Comment by Kajol-Kumari** [comment]
@Sanji515  Please take a look and if any changes is required please let me know.

**Comment by Suryansh5545** [comment]
@Kajol-Kumari you need to add node-sass as there is no other way the build will success

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari you need to add node-sass as there is no other way the build will success

Yes @Suryansh5545. I am just trying if there is any other way to deal with it. If i couldn't find any, then will add node-sass in package.json.

File: src/app/components/publiclists/challengelist/challengelist.component.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+section {
```
Comment (Sanji515) [inline src/app/components/publiclists/challengelist/challengelist.component.css]: Hey @Kajol-Kumari Why you need this `css` file? I think you should delete this file?

Comment (Kajol-Kumari) [reply src/app/components/publiclists/challengelist/challengelist.component.css]: @Sanji515 Deleted the file  as it was automatically generated by my editor(VSCode).

File: package.json [hunk 2]
```diff
@@ -41,7 +41,8 @@
     "normalize.css": "^8.0.0",
     "rxjs": "^6.5.2",
     "tslib": "^1.9.0",
-    "zone.js": "^0.8.29"
+    "zone.js": "^0.8.29",
+    "node-sass": "^4.13.1"
```
Comment (Sanji515) [inline package.json]: Now please remove this change too.

Comment (Kajol-Kumari) [reply package.json]: > Now please remove this change too.

Removed @Sanji515 but the build failed.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ import { Component, OnInit, OnChanges, Input, Output, EventEmitter, SimpleChange
 import { GlobalService } from '../../../../services/global.service';
 import { ApiService } from '../../../../services/api.service';
 import { Router, ActivatedRoute } from '@angular/router';
+import { AuthService } from '../../../../services/auth.service';
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts]: You imported this here, are you going to use this service? If yes then please add this service in the providers of spec file too else please remove this change @Kajol-Kumari

Comment (Kajol-Kumari) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts]: > You imported this here, are you going to use this service? If yes then please add this service in the providers of spec file too else please remove this change @Kajol-Kumari

@Sanji515 Removed the import as it is not needed.

File: package.json (hunk 2 above)
Comment (Sanji515) [reply package.json]: I see. Well it should work, it should install `node-sass` as a required dependency while installing other dependencies of `package.json`. Can you try changing this line [here](https://github.com/Cloud-CV/EvalAI-ngx/blob/master/.travis.yml#L20) to just `npm install`?

Comment (Kajol-Kumari) [reply package.json]: > I see. Well it should work, it should install `node-sass` as a required dependency while installing other dependencies of `package.json`. Can you try changing this line [here](https://github.com/Cloud-CV/EvalAI-ngx/blob/master/.travis.yml#L20) to just `npm install`?

@Sanji515 Sir, it's still failing

Comment (Sanji515) [reply package.json]: Also can you please try disabling cache for npm? 
```
cache:
  npm: false
```

Comment (Kajol-Kumari) [reply package.json]: > Also can you please try disabling cache for npm?
> 
> ```
> cache:
>   npm: false
> ```

Along with keeping the changes in travis.yml?

Comment (Sanji515) [reply package.json]: You can revert them.

Comment (Kajol-Kumari) [reply package.json]: > You can revert them.

It builded successfully @Sanji515

Comment (Kajol-Kumari) [reply package.json]: @Sanji515 Sir, only the Dashboard's look is left to make it same as that of EvalAI. Should i include that in this PR only?

Comment (Suryansh5545) [reply package.json]: @Kajol-Kumari This pr already have lots of stuff to review.  It will be better to make another PR for that.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.spec.ts [hunk 4]
```diff
@@ -15,7 +15,7 @@ describe('TeamcardComponent', () => {
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       declarations: [ TeamcardComponent ],
-      providers: [ GlobalService, ApiService ],
+      providers: [ GlobalService, ApiService],
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.spec.ts]: Please revert this change @Kajol-Kumari

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts [hunk 5]
```diff
@@ -38,6 +38,11 @@ export class TeamcardComponent implements OnInit, OnChanges {
    */
   @Output() selectTeamCard = new EventEmitter<any>();
 
+  /**
+   * DeSelect team event
+   */
+  @Output() DeselectTeamWrapper = new EventEmitter<any>();
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts]: Please use camel case here.

File: src/app/components/utility/side-bar/side-bar.component.scss [hunk 6]
```diff
@@ -2,7 +2,7 @@
 .ev-sidebar {
   display: block;
   height: 100vh;
-  width: 220px;
+  width: 270px;
```
Comment (Sanji515) [inline src/app/components/utility/side-bar/side-bar.component.scss]: Screenshot from 2020-02-20 20-11-02


Why we need this file, `header-static` and `footer` file changes? Due to these changes UI doesn't look same on my machine as in the current version.

File: src/app/services/endpoints.service.ts [hunk 7]
```diff
@@ -91,11 +91,17 @@ export class EndpointsService {
...
-   * Edit Team Name
+   * Edit Participant Team Name
    */
   participantTeamURL(teamId) {
     return `${this.participants}participant_team/${teamId}`;
   }
+/**
+   * Edit Host Team Name
+   */
+  HostTeamURL(teamId) {
```
Comment (Sanji515) [inline src/app/services/endpoints.service.ts]: Please use camel-case.

File: src/app/components/publiclists/teamlist/teamlist.component.ts [hunk 8]
```diff
@@ -392,10 +408,16 @@ export class TeamlistComponent implements OnInit, OnDestroy {
    */
   editTeamWrapper() {
     const SELF = this;
+    let TeamUrl;
     const editTeam = (team) => {
+      if (this.isHost) {
+        TeamUrl = SELF.endpointsService.HostTeamURL(team);
+     } else {
+        TeamUrl = SELF.endpointsService.participantTeamURL(team);
+     }
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamlist.component.ts]: Can we use ternary operator here for simplicity

File: src/app/components/publiclists/teamlist/teamlist.component.ts [hunk 9]
```diff
@@ -309,7 +325,7 @@ export class TeamlistComponent implements OnInit, OnDestroy {
     const temp = self.allTeams.slice();
     for (let i = 0; i < temp.length; i++) {
       temp[i]['isSelected'] = false;
-      if (self.selectedTeam['id'] === temp[i]['id']) {
+      if (self.OtherselectedTeam['id'] === temp[i]['id']) {
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamlist.component.ts]: Same here (Camel case)

File: src/app/components/utility/side-bar/side-bar.component.scss (hunk 6 above)
Comment (Kajol-Kumari) [reply src/app/components/utility/side-bar/side-bar.component.scss]: > Screenshot from 2020-02-20 20-11-02
> 
> Why we need this file, `header-static` and `footer` file changes? Due to these changes UI doesn't look same on my machine as in the current version.

@Sanji515  Sir, The side-nav and top-nav width and look were not same as Eval AI,so i edited it in order to make it look like that of EvalAI. If not required, i will remove them, please let me know.

Comment (Sanji515) [reply src/app/components/utility/side-bar/side-bar.component.scss]: Please show the differences from the current version if there are any cause it looks good on my device without these changes.

Comment (Kajol-Kumari) [reply src/app/components/utility/side-bar/side-bar.component.scss]: > Please show the differences from the current version if there are any cause it looks good on my device without these changes.

@Sanji515 Please take a look at the attached screenshot and if previous UI looks better, please let me know.I will reverse these changes then.

Comment (Sanji515) [reply src/app/components/utility/side-bar/side-bar.component.scss]: The only difference which I notice is that there is no vertical scroll on the sidebar right? What's your screen resolution?

Comment (Kajol-Kumari) [reply src/app/components/utility/side-bar/side-bar.component.scss]: > The only difference which I notice is that there is no vertical scroll on the sidebar right? What's your screen resolution?

And also the width of the Top-nav bar including the size of the button and texts. My Screen Resolution is 1366 x 768

Comment (Sanji515) [reply src/app/components/utility/side-bar/side-bar.component.scss]: Please remove these changes.

Comment (Kajol-Kumari) [reply src/app/components/utility/side-bar/side-bar.component.scss]: > Please remove these changes.

okay @Sanji515

**Comment by Kajol-Kumari** [comment]
Done all the changes @Sanji515 sir.

**Comment by Sanji515** [comment]
Screenshot from 2020-03-01 22-32-50

(^) doesn't look same as in the current version [here](https://evalai.cloudcv.org/web/teams) like text size, font-weight, alignment and similar other things, please check @Kajol-Kumari

**Comment by Kajol-Kumari** [comment]
> Screenshot from 2020-03-01 22-32-50
> 
> (^) doesn't look same as in the current version [here](https://evalai.cloudcv.org/web/teams) like font-weight of text, alignment and similar other things, please check @Kajol-Kumari

@Sanji515 Sir, I have made certain changes which includes font-weight,card margin and icon-size.But regarding alignment do you want the margins same as EvalAI, the it would increase the width and height of the card to some extent as there the width of side-nav and top-nav is managing that margin and padding of the team card.
Also if other changes are required, please let me know sir.

**Comment by RishabhJain2018** [comment]
@Sanji515 Can you please review this again?

**Review by Sanji515 — COMMENTED** [review]
It still doesn't look same to me as in the current version. There's a extra spacing over the host team card, it have large font-size compared to current version and also the right side icons (add member, edit and delete team) are not aligned properly which doesn't look good. Please take a look once @Kajol-Kumari

**Comment by Kajol-Kumari** [comment]
> It still doesn't look same to me as in the current version. There's a extra spacing over the host team card, it have large font-size compared to current version and also the right side icons (add member, edit and delete team) are not aligned properly which doesn't look good. Please take a look once @Kajol-Kumari

@Sanji515 can you please let me know if it looks fine.

Participant's list:
Screenshot from 2020-05-21 19-55-43

Host team list:
Screenshot from 2020-05-21 19-55-40

File: src/app/components/publiclists/challengelist/challengelist.component.scss [hunk 10]
```diff
@@ -0,0 +1,3 @@
+section{
```
Comment (Sanji515) [inline src/app/components/publiclists/challengelist/challengelist.component.scss]: Minor: Please give a space after the class name here.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss [hunk 11]
```diff
@@ -124,3 +127,20 @@
...
+
+.fa-users {
+  padding: 0px 3px 0px 8px;
+}
+
+#TeamToggleButton {
+  margin-bottom: 13.4px;
+}
+
+.fa-user-plus, .fa-pencil, .fa-trash{
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.scss]: Same here.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts [hunk 12]
```diff
@@ -38,6 +38,11 @@ export class TeamcardComponent implements OnInit, OnChanges {
    */
   @Output() selectTeamCard = new EventEmitter<any>();
 
+  /**
+   * DeSelect team event
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts]: Correct this comment here to `Deselect team event`

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts [hunk 13]
```diff
@@ -99,12 +109,15 @@ export class TeamcardComponent implements OnInit, OnChanges {
    * Select a team toggle.
    */
   selectToggle() {
-    if ((this.isHost && !this.isSelected) || !this.isHost) {
+    this.isSelected = !this.isSelected;
+    this.team['isSelected'] = this.isSelected;
+    this.deselectTeamWrapper.emit(this.team);
+  }
+
+  selectTeamToggle() {
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts:145]: Please add a comment describing this function

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts [hunk 14]
```diff
@@ -99,12 +109,15 @@ export class TeamcardComponent implements OnInit, OnChanges {
    * Select a team toggle.
    */
   selectToggle() {
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.ts]: Can we change this function name to make more sense?

File: src/app/components/publiclists/teamlist/teamlist.component.html [hunk 15]
```diff
@@ -9,12 +9,12 @@
             <app-teamcard *ngFor="let team of filteredTeams"
                           [team]="team" [selected]="team['isSelected']" [isOnChallengePage]="isOnChallengePage"
                           (deleteTeamCard)="deleteTeamWrapper()($event)" (editTeamCard)="editTeamWrapper()($event)" (addMembersTeamCard)="addMembersToTeamWrapper()($event)"
-            (selectTeamCard)="selectTeamWrapper()($event)">
+            (selectTeamCard)="selectTeamWrapper()($event)" (DeselectTeamWrapper)="DeselectTeamWrapper()($event)">
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamlist.component.html]: Here you need to change to `deselectTeamWrapper` in order to get [this](https://github.com/Cloud-CV/EvalAI-ngx/pull/289/files#diff-39040dd12941e102fae3dbad153f9262R44) working.

Also change the variable name to `deselectTeamCard` similar to `selectTeamCard`

File: src/app/components/publiclists/teamlist/teamlist.component.ts [hunk 16]
```diff
@@ -309,8 +325,8 @@ export class TeamlistComponent implements OnInit, OnDestroy {
     const temp = self.allTeams.slice();
     for (let i = 0; i < temp.length; i++) {
       temp[i]['isSelected'] = false;
-      if (self.selectedTeam['id'] === temp[i]['id']) {
-        temp[i]['isSelected'] = true;
+      if (self.otherSelectedTeam['id'] === temp[i]['id']) {
+        temp[i]['isSelectOtherselectedTeamed'] = true;
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamlist.component.ts]: I think the logic here doesn't seems right to me, also you are not using `isSelectOtherselectedTeamed`  key anywhere else. You are just assigning this key to `true` but I don't see for the `false`. I think you need to toggle this variable instead of setting directly to `true`.

Also please change the variable name

File: src/app/components/publiclists/teamlist/teamlist.component.ts [hunk 17]
```diff
@@ -297,6 +302,17 @@ export class TeamlistComponent implements OnInit, OnDestroy {
...
     const selectTeam = (team) => {
       SELF.selectedTeam = team;
+    };
+    return selectTeam;
+  }
+
+  /**
+   * Unselect a team.
+   */
+  DeselectTeamWrapper() {
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamlist.component.ts]: Please camel-case here.

File: src/app/components/publiclists/teamlist/teamlist.component.ts [hunk 18]
```diff
@@ -119,6 +119,11 @@ export class TeamlistComponent implements OnInit, OnDestroy {
    */
   selectedTeam: any = null;
 
+  /**
+   * Currently selected team to unselect
+   */
+  otherSelectedTeam: any = null;
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamlist.component.ts]: Please change the comment and variable name as I think this is the variable holding the current clicked team. Please let me know if I'm wrong @Kajol-Kumari

File: src/app/components/publiclists/teamlist/teamlist.component.ts (hunk 16 above)
Comment (Kajol-Kumari) [reply src/app/components/publiclists/teamlist/teamlist.component.ts]: @Sanji515 I think we don't need this variable as the `isSelect` will do the job here of `deSelecting` the previously selected team and `selecting` the new team.

**Comment by RishabhJain2018** [comment]
@Kajol-Kumari Can you please resolve the conflicts here?

**Review by Sanji515 — COMMENTED** [review]
@Kajol-Kumari The expected behaviour it should close all other opened team card if we select some other team but currently it is not happening can you please look at it

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari The expected behaviour it should close all other opened team card if we select some other team but currently it is not happening can you please look at it

Thank you @Sanji515 for pointing this out. I had completely missed it. I have fixed this now and attaching a GIF for the same. please check this on your side once if it looks fine.

File: src/app/components/dashboard/dashboard-content/dashboard-content.component.scss [hunk 19]
```diff
@@ -3,16 +3,4 @@
...
-
-.btn {
-  font-size: 1.2rem;
-  padding-top:.5%;
-  height: 45px;
-}
-
-.text-dark-black {
-  font-size: 3rem;
-}
```
Comment (Sanji515) [inline src/app/components/dashboard/dashboard-content/dashboard-content.component.scss:17]: Why these changes in this PR @Kajol-Kumari

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html [hunk 20]
```diff
@@ -1,41 +1,40 @@
...
-      </span>
-      <br>
-      <span class="title">
-	    Created by:
-      </span>
-      <span>
-	    {{teamView['created_by']}}
-      </span>
+    <div class="team-content" (click)='selectTeamContentToggle()'>
+      <input type="radio" class="with-gap selectPhase" name="teamSelection" (click)='selectTeamToggle()' *ngIf="isHost" id="{{teamView['team_name']}}">
```
Comment (Sanji515) [inline src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: Hey @Kajol-Kumari You also need to add the condition for if its on challenge page otherwise radio button will not be shown on the participate tab of challenge page.

File: src/app/components/dashboard/dashboard-content/dashboard-content.component.scss (hunk 19 above)
Comment (Kajol-Kumari) [reply src/app/components/dashboard/dashboard-content/dashboard-content.component.scss:17]: As due to issue with my browser's view, i had made these changes wrong in the dashboard. It's for reverting those changes @Sanji515

Comment (Sanji515) [reply src/app/components/dashboard/dashboard-content/dashboard-content.component.scss:17]: I see. Try to make separate PR for different issues but as you have added here now, lets merge with these changes too.

File: src/app/components/publiclists/teamlist/teamcard/teamcard.component.html (hunk 20 above)
Comment (Kajol-Kumari) [reply src/app/components/publiclists/teamlist/teamcard/teamcard.component.html]: yes @Sanji515 added the required condition.

**Review by Sanji515 — APPROVED** [review]
Thanks @Kajol-Kumari for all the changes. This can be merged now :tada:

**Comment by RishabhJain2018** [comment]
@Kajol-Kumari Can you resolve the conflicts here?

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari Can you resolve the conflicts here?

Done @RishabhJain2018

**Comment by RishabhJain2018** [comment]
Thank you for fixing it so quickly 🎉
